.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,39 @@
+name: CodeQL Analysis
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 8 * * *'
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: windows-latest
+    steps:
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.0.2
+
+      - name: Checkout repository
+        id: checkout_repo
+        uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        id: init_codeql
+        uses: github/codeql-action/init@v1
+        with:
+          language: csharp
+          queries: security-and-quality
+
+      - name: Build solution
+        id: build_solution
+        run: |
+          nuget restore .\TestFx.sln
+          msbuild .\TestFx.sln /p:UseSharedCompilation=false
+
+      - name: Perform CodeQL Analysis
+        id: analyze_codeql
+        uses: github/codeql-action/analyze@v1
```

Comment:
Can you exclude `~/test` folder from analysis?

Reply:
CodeQL doesn't have the option to exclude folders from the analysis for compiled languages. The suggestion in their documentation is to implement this within the build.

The solution we came up with is during the execution of the run, before the build of solution file is started, all the test projects to be removed from the solution file. This results in only the `src` and one `sample` cs-projects to be built and [CodeQL analyzing only them](https://github.com/aleks-ivanov/testfx/actions/runs/936125970).

The solution looks like this:
```
Get-ChildItem -Path test/ -Include *.csproj, *.fsproj -Recurse | ForEach-Object { 
    dotnet sln .\TestFx.sln remove $_.FullName 
}
```

Which results in these project be left in the solution file:
```
src\Adapter\MSTest.CoreAdapter\MSTest.CoreAdapter.csproj
src\Adapter\PlatformServices.Interface\PlatformServices.Interface.csproj
src\Adapter\PlatformServices.Portable\PlatformServices.Portable.csproj
src\Adapter\PlatformServices.Universal\PlatformServices.Universal.csproj
src\TestFramework\MSTest.Core\MSTest.Core.csproj
src\Adapter\PlatformServices.Desktop\PlatformServices.Desktop.csproj
src\Adapter\PlatformServices.Shared\PlatformServices.Shared.shproj
src\TestFramework\Extension.Desktop\Extension.Desktop.csproj
src\TestFramework\Extension.Core\Extension.Core.csproj
src\TestFramework\Extension.Shared\Extension.Shared.shproj
src\TestFramework\Extension.UWP\Extension.UWP.csproj
samples\FxExtensibility\FxExtensibility.csproj
src\Adapter\PlatformServices.NetCore\PlatformServices.NetCore.csproj
src\TestFramework\Extension.WinUI\Extension.WinUI.csproj
src\Adapter\PlatformServices.WinUI\PlatformServices.WinUI.csproj
```